pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -543,14 +550,14 @@ private void processStreamEvents(MessageBatch messagesAndOffsets, long idlePipeS
 
       _currentOffset = messagesAndOffsets.getNextStreamParitionMsgOffsetAtIndex(index);
       _numRowsIndexed = _realtimeSegment.getNumDocsIndexed();
-      _numRowsConsumed++;
       streamMessageCount++;
+      NUM_ROWS_CONSUMED_UPDATER.incrementAndGet(this);
```

Comment:
Concurrency bug: non atomic operation on a volatile variable with undefined results. I believe this may have resulted in the wrong number of rows in segment metadata but I haven't tracked where this value goes further than `SegmentCompletionProtocol`